tree.go

```diff
@@ -97,8 +99,13 @@ func New(h hash.Hash) *Tree {
 
 // Prove creates a proof that the leaf at the established index (established by
 // SetIndex) is an element of the Merkle tree. Prove will return a nil proof
-// set if used incorrectly. Prove does not modify the Tree.
+// set if used incorrectly. Prove does not modify the Tree. Prove can only be
+// called if SetIndex has been called previously.
 func (t *Tree) Prove() (merkleRoot []byte, proofSet [][]byte, proofIndex uint64, numLeaves uint64) {
+	if !t.proofTree {
+		panic("wrong usage: can't call prove on a tree if SetIndex wasn't called")
```

Comment:
This should be an error, don't think we need a panic.

Reply:
I feel like this is a severe programming mistake and misuse of the library and should therefore not be allowed. Using `panic` also means we don't need to break the signature.